android/src/main/java/com/hoxfon/react/RNTwilioVoice/TwilioVoiceModule.java

```diff
@@ -618,6 +641,41 @@ public void onComplete(@NonNull Task<InstanceIdResult> task) {
                 });
     }
 
+    /*
+     * Unregister your android device with Twilio
+     *
+     */
+
+    @ReactMethod  //
+    public void unregister(Promise promise) {
+        unregisterForCallInvites();
+        WritableMap params = Arguments.createMap();
+        params.putBoolean("initialized", false);
+        promise.resolve(params);
```

Comment:
Wouldn't you want to resolve the promise once `onComplete` is triggered in `unregisterForCallInvites`?

Reply:
Jesse, you can fix this part, if you think it would be correct (or more accurate), as you know the library in more depth than I am.